src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestIndexPolicyAction.kt

```diff
@@ -140,6 +141,8 @@ class RestIndexPolicyAction(
         }
 
         private fun putPolicy() {
+            newPolicy.copy(schemaVersion = IndexUtils.indexManagementSchemaVersion)
```

Comment:
the mapping logic does not need to be that complicated. All of the logic is happening in `initIndexStateManagementIndex`.

1. **does not exist -> create index w/ mappings**  This is current behavior.
2. **exists w/ lower schema number -> update index w/ new mappings**  This is new implemented behavior.
3. **exists w/ higher schema number -> error? when would this happen, only if downgrading plugin version?** In this case we do not do anything. No error and no update. We just try to use existing higher schema version. This can happen if some node has higher version of ISM plugin but should be highly unlikely.
4. **exists w/ same schema number -> proceed with normal logic** This is current behavior.
5. **exists w/ no schema number -> error? somehow dynamic mappings were created** No current json does not have schema number so it is considered `NO_SCHEMA_VERSION` and compare with new schema version.